v2/metrics.go

```diff
@@ -0,0 +1,66 @@
+package raftboltdb
+
+import (
+	"context"
+	"time"
+
+	metrics "github.com/armon/go-metrics"
+	"go.etcd.io/bbolt"
+)
+
+const (
+	defaultMetricsInterval = 5 * time.Second
+)
+
+func (b *BoltStore) RunMetrics(ctx context.Context, interval time.Duration) {
+	if interval == 0 {
+		interval = defaultMetricsInterval
+	}
+	
+	tick := time.NewTicker(interval)
+	defer tick.Stop()
+	
+	stats := b.emitMetrics(nil)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-tick.C:
+			stats = b.emitMetrics(stats)
+		}
+	}
+}
+
+func (b *BoltStore) emitMetrics(prev *bbolt.Stats) *bbolt.Stats {
+	newStats := b.conn.Stats()
+
+	stats := newStats
+	if prev != nil {
+		stats = newStats.Sub(prev)
+	}
+
+	// freelist metrics
+	metrics.SetGauge([]string{"raft", "boltdb", "numFreePages"}, float32(newStats.FreePageN))
+	metrics.SetGauge([]string{"raft", "boltdb", "numPendingPages"}, float32(newStats.FreePageN))
```

Comment:
Is there a typo here? This appears to be emitted the same value as two names. Should one of these be `stats.FreePageN` ?

Reply:
Yes there is. Its fixed now.